models/ModelSet.py

```diff
@@ -361,133 +289,36 @@ def dense(x, f, rates, conv_param):
-            pos = Reshape(target_shape=(1, 1, 1, 3))(in_pos)
-            if config['pos_noise_stdv'] != 0: pos = GaussianNoise(config['pos_noise_stdv'])(pos)
-            pos = UpSampling3D(size=x.shape[1:4])(BatchNormalization()(pos))
-            x = Concatenate(axis=-1)([x, pos])
-            for f in pos_filters:
-                pos = block(f, 4, 2, order=['dc', 'b', 'r'], order_param=[conv_param, None, None])(pos)
-                list_pos.append(pos)
-
-        for index, (sk, f) in enumerate(reversed(list(zip(skip_layer[:-1], filters[:-1])))):
-            x = block(f, 4, 2, order=['dc', 'b', 'r'], order_param=[conv_param, None, None])(x)
-            if config['feed_pos']:
-                x = tf.concat([x, sk, list_pos[index]], axis=-1)
-            else:
-                x = tf.concat([x, sk], axis=-1)
-
-        x = block(filters[0], 4, 2, order=['dc', 'b', 'r'], order_param=[conv_param, None, None])(x)
-        out = block(config['channel_label_num'], 4, 1, order=['c', 'b', 's'], order_param=[conv_param, None, None])(x)
-
-        if config['feed_pos']:
-            return create_and_compile_model([inputs, in_pos], out, config)
-        else:
-            return create_and_compile_model(inputs, out, config)
-
-    # UNet with attention focusing
-    def model_U_net_attention(self, config):
-
-        "Experimental"
-        conv_param = config['convolution_parameter']
-        inputs = Input(shape=(*config['patch_size'],) + (config['channel_img_num'],), name='inp1')
-        x = inputs
-        filters = [config['filters'] * 2 ** i for i in range(5)]
-        skip_layer = []
-        for f in filters:
-            x = block(f, 4, 2, order=['c', 'b', 'r'], order_param=[conv_param, None, None])(x)
-            skip_layer.append(x)
-        x = skip_layer[-1]
-
-        in_pos = None
-        if config['feed_pos']:
-            in_pos = Input(shape=(3,), name='input_position')
-            pos = Reshape(target_shape=(1, 1, 1, 3))(in_pos)
-            if config['pos_noise_stdv'] != 0: pos = GaussianNoise(config['pos_noise_stdv'])(pos)
-            pos = UpSampling3D(size=x.shape[1:4])(BatchNormalization()(pos))
-            x = Concatenate(axis=-1)([x, pos])
-
-        for sk, f in reversed(list(zip(skip_layer[:-1], filters[:-1]))):
-            x = block(f, 4, 2, order=['dc', 'b', 'r'], order_param=[conv_param, None, None])(x)
-            x = attention_layer_1(f, conv_param=conv_param, filters=16, alpha=1)(x, sk)
-
-        x = block(filters[0], 4, 2, order=['dc', 'b', 'r'], order_param=[conv_param, None, None])(x)
-        out = block(config['channel_label_num'], 4, 1, order=['c', 'b', 's'], order_param=[conv_param, None, None])(x)
-
-        if config['feed_pos']:
-            return create_and_compile_model([inputs, in_pos], out, config)
-        else:
-            return create_and_compile_model(inputs, out, config)
-
-    def model_U_net_2double_decoder(self, config):
-        """2 cascade U Net double decoder"""
```

Comment:
were the UNet models merged in simpler implementation? If so interfaces of these old models should be kept for backward comp. and pointing to new implementations

Reply:
model_U_net_2double_decoder can't improve the result quality.  I left only model_U_Net, and delete the variations of different U_Nets.